ENG-2364 - Add Various User API fields (#3599)
FusionAuth/fusionauth-site: astro/src/content/docs/apis/_import-users-request-body.mdx

Reviewer: This diff got wacky. Is it just the new `phoneNumber` and comment on `mobilePhone` that changed?

Author: Yes. and eslint applied its formatting opinions.

  The [instant](/docs/reference/data-types#instants) when the user was created.
</APIField>

<APIField name="users[x].lastName" type="String" optional>
  The User's last name.
</APIField>

<APIField name="users[x].memberships" type="Array" optional>
  The list of memberships for the User.
</APIField>

<APIField name="users[x].memberships[x].data" type="Object" optional>
  An object that can hold any information about the User for this membership that should be persisted.
</APIField>

<APIField name="users[x].memberships[x].groupId" type="UUID" required>
  The Id of the Group of this membership.
</APIField>

<APIField name="users[x].memberships[x].id" type="UUID" optional defaults="secure random UUID">
  The unique Id of this membership. If not specified a secure random UUID will be generated.
</APIField>

<APIField name="users[x].middleName" type="String" optional>
  The User's middle name.
</APIField>

<APIField name="users[x].mobilePhone" type="String" optional>
  The User's mobile phone number. This is useful is you will be sending push notifications or SMS messages to the User.

  {/* more */}

  <Aside>

>     This differs from <InlineField>users\[x].phoneNumber</InlineField> and cannot be used to login.

Reviewer: `login` is a noun. `log in` is a verb.
```suggestion
    This differs from <InlineField>users\[x].phoneNumber</InlineField> and cannot be used to log in.
```

Author: Fixed

  Indicates that the User's password needs to be changed during the next login attempt.
</APIField>

<APIField name="users[x].passwordLastUpdateInstant" type="Long" optional defaults="now">
  Indicates that last [instant](/docs/reference/data-types#instants) the password was changed.

  If `encryptionScheme` is omitted this value will be ignored and set to now.

  Note that if you have enabled password expiration or plan to do so in the future and you set this value to `0` or some other arbitrary time in the past that is greater than the password expiration value in days, these users will be required to change their password. It is recommended that you omit this value and allow the default value to be set unless you know the last time the user changed their password.
</APIField>

<APIField name="users[x].phoneNumber" type="String" optional since="1.99.9">
  The phone number of the User. The phone number is stored and returned in E.164 canonical format, however a phone number is considered unique regardless of the format. `303-555-1212` is considered equal to `+13035551212` so either version of this phone number can be used whenever providing it as input to an API.

>   If phoneNumber is not provided, then <InlineField>email</InlineField> or <InlineField>phoneNumber</InlineField> will be required.

Reviewer: I think this should be `username`. Why do we call out required identity fields here but not on `email` or `username`?

Perhaps this import request parameter should just say "The phone number of the User." and omit the remaining details like the other fields?
```suggestion
  If phoneNumber is not provided, then <InlineField>email</InlineField> or <InlineField>username</InlineField> will be required.
```

>   {/* more */}

Reviewer: Is this a note to add more info later? Does it do anything?

Author: I was following the pattern from the blogs, but looking again, it looks like that's arbitrary and defined by `excerpt_separator` in the front matter (right term?). Not sure if we need that.

Author: removed it

  <Aside>

>     This differs from <InlineField>users\[x].mobilePhone</InlineField> in that <InlineField>users\[x].phoneNumber</InlineField> can be used to login, just like an email address or a username.

Reviewer: ```suggestion
    This differs from <InlineField>users\[x].mobilePhone</InlineField> in that <InlineField>users\[x].phoneNumber</InlineField> can be used to log in, just like an email address or a username.
```

  </APIField>
  <APIField name="users[x].mobilePhone" type="String" optional>
    The User's mobile phone number. This is useful is you will be sending push notifications or SMS messages to the User.

    {/* more */}
    <Aside>

>       This differs from <InlineField>users[x].phoneNumber</InlineField> and cannot be used to login.

Author: Got it. And this way things look correct unlike the escaping technique I saw elsewhere
